src/rank_llm/analysis/response_analysis_verbose.py

```diff
@@ -1,26 +1,83 @@
 import json
+import os
 import re
-from typing import Dict, List
+import sys
+from typing import Dict, List, Tuple, Union
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+parent = os.path.dirname(SCRIPT_DIR)
+parent = os.path.dirname(parent)
+sys.path.append(parent)
+
+from rank_llm.result import Result
 
 
 class ResponseAnalyzer:
     def __init__(
         self,
-        files: List[str],
+        data: Union[List[str], List[Result]],
     ) -> None:
-        self._files = files
+        self._data = data
+
+    @staticmethod
+    def from_inline_results(results: List[Result]) -> "ResponseAnalyzer":
+        """
+        Method to create a ResponseAnalyzer instance from a list of Result objects.
+
+        Args:
+            results (List[Result]): A list of Result objects.
+
+        Returns:
+            ResponseAnalyzer: An instance of the ResponseAnalyzer.
+        """
+        return ResponseAnalyzer(data=results)
+
+    @staticmethod
+    def from_stored_files(filenames: List[str]) -> "ResponseAnalyzer":
+        """
+        Method to create to create a ResponseAnalyzer instance from a list of filenames.
+
+        Args:
+            filenames (List[str]): A list of filenames where each file contains data to be analyzed.
+
+        Returns:
+            ResponseAnalyzer: An instance of the ResponseAnalyzer.
+        """
+        data = []
+        for filename in filenames:
+            with open(filename, "r") as file:
+                file_data = json.load(file)
+                data.extend(file_data)
+        return ResponseAnalyzer(data=data)
 
-    def read_saved_responses(self) -> List[str]:
+    def read_results_responses(self) -> Tuple[List[str], List[int]]:
         """
-        Reads responses from the specified files and produces the total number of passages.
+        Reads responses from the specified list of Result objects and produces the total number of passages.
 
         Returns:
             Tuple[List[str], List[int]]: A tuple object containing a list of responses and a list of corresponding numbers of passages.
         """
         num_passages = []
         responses = []
-        for filename in self._files:
-            with open(filename) as f:
+        for result in self._data:
+            for exec_info in result.ranking_exec_summary:
+                responses.append(exec_info.response)
+                num_passage = self._get_num_passages(exec_info.prompt)
+                num_passages.append(int(num_passage))
+        return responses, num_passages
+
+    def read_saved_responses(self) -> Tuple[List[str], List[int]]:
+        """
+        Reads responses from the specified list of files and produces the total number of passages.
+
+        Returns:
+            Tuple[List[str], List[int]]: A tuple object containing a list of responses and a list of corresponding numbers of passages.
+        """
+        num_passages = []
+        responses = []
+        for result in self._data:
```

Comment:
in the saved responses mode, the execution summary is stored using the function in https://github.com/castorini/rank_llm/blob/main/src/rank_llm/result.py#L38
It is decoupled from the result. That's why the for loop is implemented the way it is in lines 24-30